### ISSUE TITLE
[auth_totp][9.0] issue with firefox?

### ISSUE THREAD

**Issue by andhit-r** [OCA/server-tools#908]
I tried TOTP on runbot. Here are my steps:
1. I generate Authentication Apps/Devices for Demo user using FreeTOTP on android
2. I activate MFA enable
3. Log out
4. I log in. After confirming MFA code, internal server error shows.
5. I refresh the browser. Invalid MFA code warning showed.

I tried with firefox first because it is my default browser. Then i tried with chromium. It worked. Can someone confirm it?

Many thanks

**Comment by dreispt** [comment]
Maybe the same issue as [this](https://github.com/OCA/server-tools/pull/703#issuecomment-315018876)?
cc @lasley @obulkin

**Comment by lasley** [comment]
Hmmm it seems the error is coming from Werkzeug. Only reproducible in Firefox...

```
2017-08-16 15:10:39,554 167 INFO openerp_test werkzeug: 172.17.42.1 - - [16/Aug/2017 15:10:39] "GET /auth_totp/login?redirect=&mfa_login_token=LhN6KbeEGAUaz17XnRy9 HTTP/1.0" 200 -
2017-08-16 15:10:55,223 167 INFO openerp_test werkzeug: 172.17.42.1 - - [16/Aug/2017 15:10:55] "POST /auth_totp/login HTTP/1.0" 500 -
2017-08-16 15:10:55,245 167 ERROR openerp_test werkzeug: Error on request:
Traceback (most recent call last):
File "/usr/local/lib/python2.7/dist-packages/werkzeug/serving.py", line 177, in run_wsgi
execute(self.server.app)
File "/usr/local/lib/python2.7/dist-packages/werkzeug/serving.py", line 167, in execute
for data in application_iter:
File "/usr/local/lib/python2.7/dist-packages/werkzeug/wsgi.py", line 691, in __next__
return self._next()
File "/usr/local/lib/python2.7/dist-packages/werkzeug/wrappers.py", line 81, in _iter_encoded
for item in iterable:
TypeError: 'Response' object is not iterable
```

**Comment by dreispt** [comment]
Well, my default browser is Firefox and my `auth_totp` tests were done with it.

**Comment by grindtildeath** [comment]
I also get an Internal Server Error on Firefox, and get the same error in the logs using this module with odoo v10. However the login is successful as a refresh of server URL will display odoo main page.

**Comment by obulkin** [comment]
Thanks for the info, everyone. I've traced this issue back to inconsistent behavior in [``redirect_with_hash``](https://github.com/odoo/odoo/blob/10.0/odoo/http.py#L159), which returns a ``Response`` object when the browser is Firefox and a plain string in every other case. A fix is up for review here: https://github.com/OCA/server-tools/pull/941

**Comment by dreispt** [comment]
Fixed by #941
